src/js/program/program-controller.js

```diff
@@ -58,6 +56,7 @@ export default class ProgramController extends Eventable {
                 this._setActiveMedia(mediaController);
                 // Initialize the provider last so it's setting properties on the (newly) active media model
                 mediaController.provider.init(item);
+                model.setActiveItem(index);
```

Comment:
Setting of playlist item and index should be synchronous. This would make `setPlaylistItem()` and `setPlaylistIndex` asynchronous.

Reply:
oh nvrmind!